osu.Server.Spectator/Database/DatabaseAccess.cs

```diff
@@ -465,8 +465,10 @@ public async Task<int> GetUserRankInRoomAsync(long roomId, int userId)
             var connection = await getConnectionAsync();
 
             return await connection.QuerySingleAsync<int>(
-                "SELECT COUNT(1) + 1 FROM `multiplayer_rooms_high` WHERE `room_id` = @roomId AND `user_id` != @userId "
-                + "AND `total_score` > (SELECT `total_score` FROM `multiplayer_rooms_high` WHERE `room_id` = @roomId AND `user_id` = @userId)",
+                "WITH `user_score` AS (SELECT `total_score`, `last_score_id` FROM `multiplayer_rooms_high` WHERE `room_id` = @roomId AND `user_id` = @userId) "
```

Comment:
I've never used this style of sql before, but have confirmed it works well and I have no prettier ideas.